src/main/java/org/folio/qm/service/impl/MarcRecordsServiceImpl.java

```diff
@@ -133,4 +141,18 @@ private void setUserInfo(QuickMarc quickMarc, ParsedRecordDto parsedRecordDto) {
         });
     }
   }
+
+  private Runnable updateLinksTask(FolioExecutionContext executionContext,
+                                   QuickMarc quickMarc,
+                                   DeferredResult<ResponseEntity<Void>> updateResult) {
+    var newContext = new DefaultFolioExecutionContext(executionContext.getFolioModuleMetadata(),
+      executionContext.getAllHeaders());
+    return () -> {
+      var result = (ResponseEntity<Void>) updateResult.getResult();
```

Comment:
how do you think, does it work here?
```
if (!updateResult.hasResult()) {
        return;
 }
```

Reply:
Not really. We need to check not only for result presence but also for error.